### PR TITLE
Add proper validation to mesh config

### PR DESCRIPTION
**Pull request by howardjohn**
WIP until https://github.com/istio/api/pull/1384 lands

### CODE REVIEW

File: operator/cmd/mesh/testdata/manifest-generate/output/pilot_merge_meshconfig.golden.yaml [hunk 1]
```diff
@@ -49,9 +49,9 @@ data:
     ingressService: istio-ingressgateway
     localityLbSetting:
       enabled: true
-    mixerCheckServer: foo
+    mixerCheckServer: foo:1234
     outboundTrafficPolicy:
-      mode: ALLOW_ANY
+      mode: REGISTRY_ONLY
```
Comment (howardjohn) [inline operator/cmd/mesh/testdata/manifest-generate/output/pilot_merge_meshconfig.golden.yaml:54]: This is the primary bug fix

**Comment by richardwxn** [comment]
LGTM

File: operator/pkg/component/component.go [hunk 2]
```diff
@@ -27,15 +27,15 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
 	"istio.io/api/operator/v1alpha1"
+	"istio.io/pkg/log"
+
```
Comment (ostromart) [inline operator/pkg/component/component.go:31]: nit: remove blank line.

File: operator/pkg/validate/validate.go [hunk 3]
```diff
@@ -18,15 +18,22 @@ import (
 	"fmt"
 	"reflect"
 
+	"github.com/ghodss/yaml"
 	"istio.io/api/operator/v1alpha1"
+
```
Comment (ostromart) [inline operator/pkg/validate/validate.go:23]: reformat

File: operator/pkg/validate/validate.go [hunk 4]
```diff
@@ -18,15 +18,22 @@ import (
...
 	"istio.io/istio/operator/pkg/name"
 	"istio.io/istio/operator/pkg/util"
+	"istio.io/istio/pkg/config/mesh"
 )
 
 var (
 	// DefaultValidations maps a data path to a validation function.
 	DefaultValidations = map[string]ValidatorFunc{
+		"Values": func(path util.Path, i interface{}) util.Errors {
+			return CheckValues(i)
```
Comment (ostromart) [inline operator/pkg/validate/validate.go:34]: nice change, thanks.

**Comment by ostromart** [comment]
overall looks good.

File: operator/pkg/component/component.go (hunk 2 above)
Comment (howardjohn) [reply operator/pkg/component/component.go:31]: This is done by `make format`. I have no control

File: operator/pkg/validate/validate.go (hunk 3 above)
Comment (howardjohn) [reply operator/pkg/validate/validate.go:23]: Actually I might have not run `make format` . Oops

**Comment by howardjohn** [comment]
Blocked by the API update breaking CRDs (unrelated to this change)

**Review by ostromart — APPROVED** [review]

**Review by linsun — APPROVED** [review]

**Pull request merged**